Don't search host directory during "relink" if $inst_prefix is provided.
Taken from Libtool recipe's patches, for instance libtool-2.4/inst-prefix-
only.patch.
Index: SDL-1.2.14/build-scripts/ltmain.sh
===================================================================
--- SDL-1.2.14.orig/build-scripts/ltmain.sh
+++ SDL-1.2.14/build-scripts/ltmain.sh
@@ -5786,12 +5791,14 @@ func_mode_link ()
 	      fi
 	    else
 	      # We cannot seem to hardcode it, guess we'll fake it.
-	      add_dir="-L$libdir"
-	      # Try looking first in the location we're being installed to.
+	      
+	      # Default if $libdir is not relative to the prefix:
+	      add_dir="-L$libdir"
+	      
 	      if test -n "$inst_prefix_dir"; then
 		case "$libdir" in
                   [\\/]*)
-                    add_dir="$add_dir -L$inst_prefix_dir$libdir"
+                    add_dir="-L$inst_prefix_dir$libdir"
                     ;;
                 esac
               fi
